src/core/tsi/s2a/record_protocol/s2a_crypter.cc

```diff
@@ -256,3 +266,227 @@ void check_half_connection(s2a_crypter* crypter, bool in_half_connection,
   GPR_ASSERT(half_connection->additional_data_size ==
              expected_additional_data_size);
 }
+
+/** This function returns the max number of bytes occupied by the nonce of a
+ *  TLS 1.3 record that is handled by |crypter|. The caller must not pass in
+ *  nullptr for |crypter|.
+ *  - crypter: an instance of s2a_crypter. **/
+static size_t s2a_max_aead_nonce_size(const s2a_crypter* crypter) {
+  GPR_ASSERT(crypter != nullptr);
+  /** If additional supported ciphersuites are added, then there may be
+   *  additional options for the return value. **/
+  return kEvpAeadMaxNonceLength;
```

Comment:
It seems a little weird in a sense that `kEvpAeadMaxNonceLength` is not a part of `s2a_crypter`. Do we need this function?

Reply:
I agree it looks a little strange right now, but it's because we might add other ciphersuites or other TLS versions, in which case the max nonce length would change. I wrote it this way so the API would be stable.